Cargo.toml

```diff
@@ -4,7 +4,7 @@ resolver = "2"
 
 [patch.crates-io]
 rocket = { git = "https://github.com/SergioBenitez/Rocket" } # Need to patch rocket dependency of `rocket_basicauth` until there is an official release.
-xtra = { git = "https://github.com/Restioson/xtra" } # Latest master has crucial patches.
+xtra = { git = "https://github.com/Restioson/xtra", rev = "413dc807bd70d70cfe4a02cb3fd7d49d10d827e9" } # Latest master has crucial patches.
```

Comment:
Did a `cargo update -p xtra` not suffice?

IMO the lockfile should be enough here, we don't need to duplicate this information.

Reply:
I find it misleading to imply "latest" master here and then discover in the `lockfile` that we actually point to a specific revision that is not necessarily the latest one.

But if you feel strongly I can remove it.

Reply:
Even with your patch, we are not guaranteed that we point to latest master :)

Feel free to change the comment to something like: "We want to utilize unreleased patches." if you find that that is clearer!

Reply:
What I mean is that in this case (probably in most cases) we don't want to point to the moving target that is latest master. When this was added we wanted to include some specific commits which happened to include latest master at the time. And that was information was recorded in the lockfile.

My preference would be to always specify the revision here, so that one doesn't have to go to the lockfile (which one never edits manually) to find out what we are actually depending on. AFAIK this duplicated information cannot get out of sync. If it can then I would accept your suggestion and remove this.

Reply:
A matter of taste I guess :)

In case you didn't know, `cargo` has a `precise` flag to update a dependency to a particular version, so you could run:

```
cargo update -p xtra --precise 413dc807bd70d70cfe4a02cb3fd7d49d10d827e9
```

I agree that the lockfile should not be edited manually.

Reply:
Actually, something really weird happens in the lockfile using `rev`. Currently on this branch:
```
source = "git+https://github.com/Restioson/xtra?rev=413dc807bd70d70cfe4a02cb3fd7d49d10d827e9#413dc807bd70d70cfe4a02cb3fd7d49d10d827e9"
```
After running `cargo update -p xtra --precise 224004bc95dac12b8b8d2a14fea5ddf37eef6ae0`, actually pointing to latest master:

```
source = "git+https://github.com/Restioson/xtra?rev=413dc807bd70d70cfe4a02cb3fd7d49d10d827e9#224004bc95dac12b8b8d2a14fea5ddf37eef6ae0"
```

The patch information is duplicated and the one after the hash seems to take precedence. I think I'm gonna do as you originally suggested.

Reply:
That is indeed really weird!